### ISSUE TITLE
[V7] Request for Quotation Suggestions

### ISSUE THREAD

**Issue by bcornwellmott** [frappe/erpnext#5493]
I have been looking at integrating the Request for Quotation functionality and have noticed some issues:

1. There doesn't seem to be a way to generate RFQs for suppliers without actually sending an email to them. Some suppliers that we work with will not respond via email, or might not want to access the portal. _Just being able to create a supplier specific PDF would be useful_. This ties in with: 
2. The Manufacturers Part Number only shows up in the Portal. It is not shown on the PDF emailed to the supplier, and supplier specific PDFs cannot be generated. 
3. Re-sending Email. If one of my suppliers asks me to resend the RFQ email, there is no way to do that without sending an email to every supplier. _Being able to select the supplier to send an email would be useful._
4. We cannot attach documents to the emails sent to suppliers. Since our company designs many of our own parts, we have drawings and schematics that need to be attached to the Request for Quotations we send to suppliers. Right now, the email sends a list of what items needs to be quoted to the supplier, but there is no way for us to send our suppliers the details of what each of those items is. _Being able to attach files and have them additionally attached to the email is critical._
5. _Suppliers cannot attach documents to the quotation_. Some of our suppliers will want to attach their own documents when responding to the quotes. This could be because they want to supply us with an "official" document of the quotation to avoid controversy later on, or any other reason. 
6. The Currency is not shown on the supplier portal. Since many suppliers work in multiple countries or might quote in multiple currencies, they should either be told what currency to quote in or asked to select the quotation currency. Right now, there is no indication of the currency anywhere on the portal page.

To summarize, the big things for us using it are not being able to attach documents to the email being sent, not being able to generate custom PDFs for each supplier, and not being able to select which suppliers get emails.

Thank you!

**Comment by kodybaker** [comment]
Agreed!

**Comment by rmehta** [comment]
@bcornwellmott  this is good feedback!

@rohitwaghchaure can you try and incorporate these?

**Comment by rohitwaghchaure** [comment]
@bcornwellmott 
Few modifications are as follow:-

1. User can resend email to suppliers, on supplier table unchecked send email to whom you don't want to resend an email.

1. Part Number and supplier name are available on print format which will sent to supplier.

1. If user want to send attachment to supplier, attach document to RFQ.

[https://www.youtube.com/watch?v=1snON19SSx8](url)

**Comment by bcornwellmott** [comment]
The video is showing as private, but the description of the changes is great!

**Comment by rohitwaghchaure** [comment]
Change it to public, please check 
https://www.youtube.com/watch?v=1snON19SSx8

**Comment by bcornwellmott** [comment]
These are really good changes! 
Once this is merged with the develop branch I will give it a test run and see if there is anything else we notice, but you've covered the really big ones.

One thing wasn't clear from the video... Is there a way to download the pdf that is sent to a specific supplier?

**Comment by rohitwaghchaure** [comment]
@bcornwellmott 
Changes are merged into develop branch please check, I am working on feature to download pdf for specific supplier from RFQ form.

**Comment by rohitwaghchaure** [comment]
@bcornwellmott 
Done with download pdf, please check below video
https://youtu.be/Z5nkQNmaW7Q

**Comment by bcornwellmott** [comment]
This is really good! Thank you for implementing so much of our feedback.

We noticed a few issues when we were going through a test run:
1. The currency is still not showing up in the Supplier portal.
2. There is no way for suppliers to add Shipping / Taxes in the supplier portal. Right now they have the ability to add it in notes, but that is less convenient and could cause mistakes.
3. Disabling the portal (in Portal Settings) does not seem to have any effect. It is still referenced in the email and the link still works. It would be nice to be able to disable the portal and still use the RFQ functionality, or better yet choose which suppliers see the portal/link.
4. I can't see a way to edit the text of some of the email that is sent:

> Request for Quotation
> 
> Please supply the specified items at the best possible rates
> 
> Request for quotation can be access by clicking following link:
> 
> Submit your Quotation
> 
> Thank you,
> Ben Cornwell-Mott
> 
> 
> 
> This email was sent automagically by VeloMetro's ERP system.
 Everything below the "Please Supply the specified items at the best possible rates" is fixed and can't be changed (from what I can tell). I could edit `erpnext/templates/emails/request_for_quotation.html` but I think that would cause issues on bench-update.

Thanks again! This is really good stuff.

**Comment by rohitwaghchaure** [comment]
@bcornwellmott 
Thanks for your suggestion,
Regarding currency, it is a major feature which will be added in next update. Supplier can not select taxes and charges template on portal because he is not aware about erpnext's tax structure. Instead he can put the details of tax in terms. Which user has to add manually on supplier quotation. I have fixed the issue of portal settings. You can not edit the below text as it is a standard.
"Request for quotation can be access by clicking following link:"
"Submit your Quotation"

**Comment by muchai** [comment]
Hi,
More suggestion, one was already mentioned by @bcornwellmott
1.	Suppliers cannot attach documents to the quotation. Some of our suppliers will want to attach their own documents when responding to the quotes. This could be because they want to supply us with an "official" document of the quotation to avoid controversy later on, or any other reason.
2.	A save button on the suppliers portal, request for quotation page, that wat, users can update prices, and save, review if need be, and submit once they’re ready.

In addition, working on an improvement, for opening rfqs, suggestions are welcome. Here's the draft flow: https://github.com/frappe/erpnext/issues/16351
